docs/src/index.md

```diff
@@ -0,0 +1,457 @@
+@select(df, :x2 = :x, :x3 = ^(:x))
+```
+
+This rule applies to all DataFramesMeta macros.
+
+## Comparison with `dplyr` and LINQ
+
+A number of functions for operations on DataFrames have been defined.
+Here is a table of equivalents for Hadley's
+[dplyr](https://github.com/hadley/dplyr) and common
+[LINQ](http://en.wikipedia.org/wiki/Language_Integrated_Query)
+functions.
+
+    Julia             dplyr            LINQ
+    ---------------------------------------------
+    @where            filter           Where
+    @transform        mutate           Select (?)
+    @by                                GroupBy
+    groupby           group_by         GroupBy
+    @combine          summarise/do
+    @orderby          arrange          OrderBy
+    @select           select           Select
+
+
+## `@linq` and other chaining macros
+
+There is also a `@linq` macro that supports chaining and all of the
+functionality defined in other macros. Here is an example of `@linq`:
+
+```julia
+df = DataFrame(a = repeat(1:5, outer = 20),
+               b = repeat(["a", "b", "c", "d"], inner = 25),
+               x = repeat(1:20, inner = 5))
+
+x_thread = @linq df |>
+    transform(y = 10 * :x) |>
+    where(:a .> 2) |>
+    by(:b, meanX = mean(:x), meanY = mean(:y)) |>
+    orderby(:meanX) |>
+    select(:meanX, :meanY, var = :b)
+```
+
+Relative to the use of individual macros, chaining looks cleaner and
+more obvious with less noise from `@` symbols. This approach also
+avoids filling up the limited macro name space. The main downside is
+that more magic happens under the hood.
+
+This method is extensible. Here is a comparison of the macro and
+`@linq` versions of `with`.
+
+```julia
+function linq(::DataFramesMeta.SymbolParameter{:with}, d, body)
+    with_helper(d, body)
+end
+
+df = DataFrame(A = 1:3, B = [2, 1, 2])
+
+@linq df |>
+    with(:A .* :B)
+```
+
+The `linq` method above registers the expression-replacement method
+defined for all `with()` calls. It should return an expression like a
+macro.
+
+Again, this is experimental. Based on feedback, we may decide to only
+use `@linq` or only support the set of linq-like macros.
+
+Alternatively you can use Lazy.jl `@>` macro like this:
+
+```julia
+using Lazy: @>
+
+df = DataFrame(a = repeat(1:5, outer = 20),
+               b = repeat(["a", "b", "c", "d"], inner = 25),
+               x = repeat(1:20, inner = 5))
+
+x_thread = @> begin
+    df
+    @transform(y = 10 * :x)
+    @where(:a .> 2)
+    @by(:b, meanX = mean(:x), meanY = mean(:y))
+    @orderby(:meanX)
+    @select(:meanX, :meanY, var = :b)
+end
+```
+
+Please note that Lazy.jl exports the function `groupby` which would clash
+with `DataFrames.groupby`. Hence, it is recommended that you only import a
+select number of functions into the namespace by only importing `@>` e.g. 
+`using Lazy: @>` instead of `using Lazy`.
+
+Another alternative is Pipe.jl which exports the `@pipe` macro for piping. 
```

Comment:
same with Pipe.jl. I like and use it with DataFrames.jl but what is the benefit here? Do we recommend it?

Reply:
Actually, I think I will keep the piping sections. `@>` and `@pipe` have different behaviors, so it makes sense for users to choose what they are most comfortable with, for now. 

Eventually I would like DataFramesMeta to provide one really good macro and not recommend others, but that hasn't happened yet.